### PR TITLE
added regexp validation support

### PR DESCRIPTION
**Pull request by tivvit**
regexp validation support for checkboxes and radiobuttons with documentation

### CODE REVIEW

File: src/CheckboxList.php [hunk 1]
```diff
@@ -304,7 +304,17 @@ public static function validateRange(CheckboxList $control, array $range)
 		return static::validateMinLength($control, $range[0]) && static::validateMaxLength($control, $range[1]);
 	}
 
-
+    /**
+     * Validator for regular expressions of checked boxes
+     *
+     * @param Checkboxlist
```
Comment (vojtech-dobes) [inline src/CheckboxList.php]: param annotation should be followed by two spaces.

File: src/CheckboxList.php [hunk 2]
```diff
@@ -304,7 +304,17 @@ public static function validateRange(CheckboxList $control, array $range)
 		return static::validateMinLength($control, $range[0]) && static::validateMaxLength($control, $range[1]);
 	}
```
Comment (vojtech-dobes) [inline src/CheckboxList.php:306]: Methods should be separated by three empty lines.

File: src/RadioList.php [hunk 3]
```diff
@@ -266,4 +266,16 @@ public static function register()
...
 	}
 
+    /**
+     * Validator for regular expressions for radioList
+     *
+     * @param RadioList
+     * @param string
+     * @return bool
+     */
+    public static function validateregexp(RadioList $control, $regexp)
```
Comment (vojtech-dobes) [inline src/RadioList.php]: Method should be camelCase, not deadcamelcase :).

File: src/RadioList.php [hunk 4]
```diff
@@ -266,4 +266,16 @@ public static function register()
...
+    {
+        return (bool) preg_match($regexp,$control->getValue());
```
Comment (vojtech-dobes) [inline src/RadioList.php]: Comma should be followed by space.

**Comment by vojtech-dobes** [comment]
Thanks for the pull request, just please fix the coding standard issues. I will merge it then. Also for next time, pull request should be opened not from `master` branch but from some specific branch.

**Comment by tivvit** [comment]
Thanks for patience and valuable advices. Next time i will do better beacause I switched on showing whitespaces and tabs in IDE ;)

**Comment by vojtech-dobes** [comment]
Thank you for contribution :). Merged in ee1c901.

**Comment by vojtech-dobes** [comment]
Released as 1.0.4.

**Pull request closed without merging**